src/vconsole/vconsole-setup.c

```diff
@@ -328,48 +320,79 @@ static void setup_remaining_vcs(int fd, bool utf8) {
         }
 }
 
+static const char *find_source_vc(const char *vc, int *src_fd, int *src_idx) {
+        static char ttyname[sizeof("/dev/tty") - 1 + DECIMAL_STR_MAX(int)];
+
+        if (vc) {
+                _cleanup_close_ int fd = -1;
+
+                fd = open_terminal(vc, O_RDWR|O_CLOEXEC);
+                if (fd < 0) {
+                        log_error_errno(fd, "Failed to open %s: %m", vc);
+                        return NULL;
```

Comment:
hmm, i think we should propagate the error as error really and not as NULL. i.e. if the function returns an int then a positive return value would be an fd, and the negative returns would be a negative errno error like we do everywhere else (also see CODING_STYLE about that)